src/gc/gc.cpp

```diff
@@ -34212,11 +34212,30 @@ HRESULT GCHeap::Initialize()
 
 #ifdef MULTIPLE_HEAPS
     nhp_from_config = static_cast<uint32_t>(GCConfig::GetHeapCount());
-    
+
     // GetCurrentProcessCpuCount only returns up to 64 procs.
-    uint32_t nhp_from_process = GCToOSInterface::CanEnableGCCPUGroups() ?
-                                GCToOSInterface::GetTotalProcessorCount():
-                                GCToOSInterface::GetCurrentProcessCpuCount();
+    uint32_t nhp_from_process;
+    if (GCToOSInterface::CanEnableGCCPUGroups())
+    {
+        nhp_from_process = GCToOSInterface::GetTotalProcessorCount();
+    }
+    else
+    {
+        uint32_t cpuCount = GCToOSInterface::GetCurrentProcessCpuCount();
+        double cpuBudget = GCToOSInterface::GetCurrentProcessCpuBudget();
+        assert (1.0 <= cpuBudget && cpuBudget < (double) UINT32_MAX);
+
+        if (cpuBudget < cpuCount)
+        {
+            cpuCount = (uint32_t)(cpuBudget + 0.5);
```

Comment:
You can use `cpuCount = (uint32_t)(cpuBudget + 0.99)` instead of having to update the cpuCount after the rounding. You can choose an arbitrary number like 0.999, 0.9999 or whatever value close to 1.